libdrgn/arch_x86_64.c

```diff
@@ -574,18 +574,49 @@ linux_kernel_pgtable_iterator_init_x86_64(struct drgn_program *prog,
 	memset(it->index, 0xff, sizeof(it->index));
 }
 
+// pte flags
+static const uint64_t _PAGE_PRESENT = 1 << 0;
+static const uint64_t _PAGE_PROTNONE = 1 << 8;
+
+// TODO: PTE_PFN_MASK may be different because of CONFIG_DYNAMIC_PHYSICAL_MASK
+static const uint64_t PTE_PFN_MASK = UINT64_C(0xffffffffff000);
+static const int PAGE_SHIFT = 12;
+
+static inline bool __pte_needs_invert(uint64_t val)
+{
+	return val && !(val & _PAGE_PRESENT);
+}
+
+static inline uint64_t protnone_mask(uint64_t val)
+{
+	return __pte_needs_invert(val) ?  ~0ull : 0;
```

Comment:
In theory, `~0ull` could be a different size than `uint64_t`, so I'd feel better about:

```suggestion
	return __pte_needs_invert(val) ? -1 : 0;
```